### PR TITLE
Bug 2014161: Fix pipeline run logs autoscrolling issue

### PR DESCRIPTION
**Pull request by karthikjeeyar**
**Fixes**: 

https://issues.redhat.com/browse/OCPBUGSM-36824

**Analysis / Root cause**: 

High volume of data streaming into the logs components, causes too many calls to scrollIntoView method. It results failure to autoscroll to the latest content and sometimes the browser goes unresponsive.

**Solution Description**: 

Added throttling to the append the content to the dom and to scroller function.

**Screen shots / Gifs for design review**: 

https://user-images.githubusercontent.com/9964343/154067572-da1d1ba0-795b-4ee1-8ecd-8d053b193eaa.mov

**Unit test coverage report**: 

No tests changed.

**Test setup:**

1. Import an application with pipeline option enabled in Import from Git flow (use: https://github.com/alexeykazakov/spring-petclinic)
2. Go to pipelinerun logs page.

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/pipelines-plugin/src/components/pipelineruns/logs/Logs.tsx [hunk 1]
```diff
@@ -34,17 +35,28 @@ const Logs: React.FC<LogsProps> = ({
...
+    if (contentRef.current) {
+      contentRef.current.innerText += blockContentRef.current;
+    }
+    if (scrollToRef.current) {
+      scrollToRef.current.scrollIntoView({ behavior: 'smooth', block: 'end' });
+    }
+    blockContentRef.current = '';
+  }, 300);
+
+  const contentScrollerRef = React.useRef<() => void>(addContentAndScroll);
```
Comment (rohitkrai03) [inline frontend/packages/pipelines-plugin/src/components/pipelineruns/logs/Logs.tsx:51]: Why do we need a ref for a throttled function?

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: frontend/packages/pipelines-plugin/src/components/pipelineruns/logs/Logs.tsx (hunk 1 above)
Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/components/pipelineruns/logs/Logs.tsx]: Ref is not needed and I have updated the code to use useCallback.

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Pull request merged**